components/merkledb/src/validation.rs

```diff
@@ -12,26 +23,43 @@ pub fn is_valid_index_name<S: AsRef<str>>(name: S) -> bool {
 /// Check that latin1 character is allowed in index name.
 /// Only these combination of symbols are allowed:
 ///
-/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`, `.`
+/// `[0..9]`, `[a-z]`, `[A-Z]`, `_`, `-`
```

Comment:
Well, dot **is** a valid latin1 character, so now this function is not really correct. I'd suggest rename it to the `is_allowed_index_name_char` or something like this.